### PR TITLE
Rename to palindrom-polymer and remove Palindrom logic

### PR DESCRIPTION
**Pull request by alshakero**
Remove Palindrom logic in favour of `<palindrom-connection />` events.

Related: https://github.com/Palindrom/PalindromTaskForce/blob/master/README.md#tasks-group-ii

### CODE REVIEW

File: palindrom-polymer.html [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+                    }
+                },
+                target: {
+                    type: String,
+                    notify: true,
+                    value: 'palindrom-connection'
+                }
+            },
+            attached: function () {
+                const palindromConnection = document.querySelector(this.target);
```
Comment (tomalec) [inline palindrom-polymer.html]: What if they are both in the same shadow Root?/
Please query the `this.getRootNode()`

File: palindrom-polymer.html [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+                                    //console.warn('remove is translated to JSON incompatible set to `undefined`, as there is no remove method in Polymer.')
+                                    templateDomBind.notifyPath(polymerPath, undefined, false);
+                                    break;
+                            }
+                        }
+                    }
+                }
+            },
+            validateJSONPatch: function () {
+                console.info('inherit validateJSONPatch from fast-json-patch');
```
Comment (tomalec) [inline palindrom-polymer.html]: is this to be implemented?

File: palindrom-polymer.html [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+            return String(n) === str && n >= 0;
+        }
+
+        return component;
+    }());
+
+</script>
+<dom-module id="palindrom-polymer">
+    <script>
+        var palindromClient = Object.assign({}, component);
```
Comment (tomalec) [inline palindrom-polymer.html]: is the `component` a global variable? please make it local

**Comment by tomalec** [comment]
On one hand I like to DOM events approach as it is clean and uses DOM as the API, but on the other I'm worried about the performance. Isn't it too expensive? Have you tested it somehow? (@mmnosek it would be great to have some kind of benchmarks for Palindrom flow)

Maybe, as you query the DOM to find exact element to bind to, you can also use more direct connection?

**Comment by alshakero** [comment]
@tomalec You're right, I **guess** events are slower by magnitudes, for instance, when I run the tests, I could see the left iframe changes quite more obviously than before. 

But:
1. The only reasonable way to bubble is using events.
2. Callbacks allow only one subscriber.

I'm open to ideas.

File: palindrom-polymer.html (hunk 1 above)
Comment (alshakero) [inline palindrom-polymer.html]: Apparently, `getRootNode` only works in Safari 10.1+, and SauceLabs has Safari 10 only. Do you have a walkaround?

**Comment by alshakero** [comment]
I think this is g2g that we agreed to use events?

File: palindrom-polymer.html [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+            return root;
+        }
+        function getRoot(node) {
+            if (node.parentNode != null) {
+                return getRoot(node.parentNode);
+            }
+
+            return node;
+        }
+        function isShadowRoot(node) {
```
Comment (tomalec) [inline palindrom-polymer.html]: Those functions are global, please make then local.

File: palindrom-polymer.html [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+                if (this.bound) {
+                    this.notifyTemplateDomBind(this.bound.model, sequence, this.bound, "model");
+                }
+                else {
+                    this.notifyTemplateDomBind(this.obj, sequence, this, "obj");
+                }
+            },
+            onLocalChange: function (ev) {
+                const patch = ev.detail;
+                this.propagateDomBind({ patch, results: patch.map(function () { }) });
```
Comment (tomalec) [inline palindrom-polymer.html]: you can use `noop`

**Comment by alshakero** [comment]
@tomalec so should we rename the repo release 5.0.0?

**Comment by alshakero** [comment]
Ping.

**Comment by warpech** [comment]
@tomalec ping

**Comment by alshakero** [comment]
@tomalec approved it, but I didn't merge because I'm using the branch in my experiments in case I discover I needed some changes throughout the process I can push them here.

**Comment by warpech** [comment]
Is this PR abandoned?

**Comment by alshakero** [comment]
Yes. We're not proceeding with this.

**Pull request closed without merging**